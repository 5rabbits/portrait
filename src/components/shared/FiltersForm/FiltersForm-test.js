import { React, toJSON, mount } from 'test/helper'
import PropTypes from 'prop-types'
import FiltersForm from 'shared/FiltersForm'

describe('FiltersForm', () => {
  it('should render with minimum props', () => {
    const component = mount(
      <FiltersForm>
        Test
      </FiltersForm>,
    )

    expect(toJSON(component)).toMatchSnapshot()
  })

  describe('if context.toggleFilters is defined', () => {
    it('should render the close filters link', () => {
      const component = mount(
        <FiltersForm>
          Test
        </FiltersForm>,
        {
          context: {
            toggleFilters: () => {},
          },
          childContextTypes: {
            toggleFilters: PropTypes.func,
          },
        },
      )

      expect(toJSON(component)).toMatchSnapshot()
    })
  })
})
