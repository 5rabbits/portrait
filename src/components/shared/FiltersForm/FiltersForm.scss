@import '~styles/core';

.FiltersForm {
  height: 100%;
  display: flex;
  flex-direction: column;

  .control-label {
    text-transform: uppercase;
    color: #94A3B1;
    font-size: 10px;
    font-weight: 700;
  }

  .Select-control,
  .form-control {
    @include transparentInput();
  }
}

.FiltersForm__title {
  font-size: 18px;
  font-weight: 700;
  margin-bottom: 16px;
}

.FiltersForm__contents,
.FiltersForm__submit {
  width: 100%;
}

.FiltersForm__contents {
  flex: 1;
  overflow: auto;
  padding-bottom: $gutter;
}

.FiltersForm__submit {
  flex-shrink: 0;
  padding-bottom: $gutter;
  position: relative;

  &:before {
    pointer-events: none;
    content: "";
    display: block;
    position: absolute;
    height: $gutter;
    top: -$gutter;
    left: 0;
    right: 12px;
    background-image: linear-gradient(-180deg, rgba($body-color, 0) 0%, $body-color 80%);
  }

  .btn {
    @include filtersButton();
  }
}

.FiltersForm__filters-toggle {
  height: $table-section-toolbar-height;
  display: flex;
  align-items: center;
}
