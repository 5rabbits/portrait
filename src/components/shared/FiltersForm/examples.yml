code: >
  <FiltersForm style={{ height: 240, width: 300 }}>
    <div className="form-group">
      <label>Field 1</label>
      <input className="form-control" />
    </div>
    <div className="form-group">
      <label>Field 2</label>
      <input className="form-control" />
    </div>
    <div className="form-group">
      <label>Field 3</label>
      <input className="form-control" />
    </div>
  </FiltersForm>
