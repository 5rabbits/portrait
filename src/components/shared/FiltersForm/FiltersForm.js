import React, { PureComponent } from 'react'
import PropTypes from 'prop-types'
import { Container } from 'shared'
import cx from 'classnames'
import './FiltersForm.scss'

/**
 * A component to be used as the advanced filters of
 * [TableContainer](/components/TableContainer).
 *
 * The search button will be fixed at the bottom of the form
 * and the filters will be placed on a scrollable container.
 */
export default class FiltersForm extends PureComponent {
  static propTypes = {
    /**
     * CSS class assigned to the form.
     */
    className: PropTypes.string,

    /**
     * The filters fields.
     */
    children: PropTypes.node.isRequired,
  }

  static defaultProps = {
    className: null,
  }

  render() {
    const { children, className, ...other } = this.props

    return (
      <form
        {...other}
        className={cx('FiltersForm', className)}
        >
        <Container className="FiltersForm__contents">
          <h4 className="type-subhead mb-sm">Filtrar resultados</h4>
          {children}
        </Container>
        <Container className="FiltersForm__submit">
          <button
            className="btn btn-block btn-primary"
            type="submit"
            >
            Buscar
          </button>
        </Container>
      </form>
    )
  }
}
