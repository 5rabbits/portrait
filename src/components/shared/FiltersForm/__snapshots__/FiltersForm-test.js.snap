// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FiltersForm if context.toggleFilters is defined should render the close filters link 1`] = `
<FiltersForm
  className={null}
>
  <form
    className="FiltersForm"
  >
    <ScrollLock
      selector={null}
    >
      <Container
        className="FiltersForm__contents"
        size="normal"
      >
        <div
          className="Container Container--normal FiltersForm__contents"
        >
          <div
            className="FiltersForm__filters-toggle"
          >
            <LayoutLink
              active={false}
              className={null}
              disabled={false}
              href="#"
              onClick={[Function]}
            >
              <a
                className="LayoutLink"
                href="#"
                onClick={[Function]}
              >
                <span>
                  <i
                    className="fa fa-times"
                  />
                   
                  Close filters
                </span>
              </a>
            </LayoutLink>
          </div>
          <h4
            className="FiltersForm__title"
          >
            Filters
          </h4>
          Test
        </div>
      </Container>
    </ScrollLock>
    <Container
      className="FiltersForm__submit"
      size="normal"
    >
      <div
        className="Container Container--normal FiltersForm__submit"
      >
        <button
          className="btn btn-block btn-lg"
          type="submit"
        >
          Filter
        </button>
      </div>
    </Container>
  </form>
</FiltersForm>
`;

exports[`FiltersForm should render with minimum props 1`] = `
<FiltersForm
  className={null}
>
  <form
    className="FiltersForm"
  >
    <ScrollLock
      selector={null}
    >
      <Container
        className="FiltersForm__contents"
        size="normal"
      >
        <div
          className="Container Container--normal FiltersForm__contents"
        >
          <h4
            className="FiltersForm__title"
          >
            Filters
          </h4>
          Test
        </div>
      </Container>
    </ScrollLock>
    <Container
      className="FiltersForm__submit"
      size="normal"
    >
      <div
        className="Container Container--normal FiltersForm__submit"
      >
        <button
          className="btn btn-block btn-lg"
          type="submit"
        >
          Filter
        </button>
      </div>
    </Container>
  </form>
</FiltersForm>
`;
